astropy/convolution/tests/test_convolve_fft.py

```diff
@@ -342,20 +343,66 @@ def test_uniform_3_withnan(self, boundary, nan_treatment,
         if nan_treatment == 'interpolate':
             answer_key += '_interpnan'
 
-        posns = np.where(np.isfinite(z))
+        posns = np.isfinite(z)
 
-        assert_floatclose(z[posns], answer_dict[answer_key][posns])
+        answer = answer_dict[answer_key][posns]
 
-    def test_nan_fill(self):
+        # check that fill is set and that the 1'th position that was originally
+        # NaN is included in the check
+        if (nan_treatment == 'fill') and posns[1]:
+            # we fill the center with the sum of the input array divided by
+            # three, since we've now pre-filled the center value with zero
+            answer[1] = 4 / (3. if normalize_kernel else 1.)
+
+        assert_floatclose(z[posns], answer)
+
+    def test_nan_interpolate(self):
 
         # Test masked array
         array = np.array([1., np.nan, 3.], dtype='float64')
         kernel = np.array([1, 1, 1])
         masked_array = np.ma.masked_array(array, mask=[0, 1, 0])
+
         result = convolve_fft(masked_array, kernel, boundary='fill',
+                              nan_treatment='interpolate',
                               fill_value=np.nan)
```

Comment:
Ok, so I've been trying to figure out the point to this test, specifically why fill_value=np.nan. In fact, I think it points out another bug. I'll open an issue.

In the meantime, what is the intent of this test and why fill_value=np.nan?

Reply:
https://github.com/astropy/astropy/issues/8127